sdk/messaging/azeventhubs/checkpointstore/blob_checkpoint_store.go

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package checkpointstore
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/internal/blob"
+)
+
+// BlobCheckpointStore is a CheckpointStore implementation that uses Azure Blob storage.
+type BlobCheckpointStore struct {
```

Comment:
I feel like the name is a little long and redundant.

`checkpointstore.BlobCheckpointStore`

`checkpointstore.Blob` would be more concise but I get that just `Blob` is a little weird.  Is there some other name we can use so that we can remove the redundant suffix?

Reply:
Blob feels good to me since it's under the checkpointstore package already. Let's go with that one.

Reply:
Hm...maybe BlobStore?

Reply:
Okay, renamed to:

checkpoints.BlobStore